static/js/whyis.js

```diff
@@ -905,56 +905,43 @@ function whyis() {
-                if (!promises[uri]) {
-                    var localPart = uri.split("#").filter(function(d) {return d.length > 0});
-                    localPart = localPart[localPart.length-1];
-                    localPart = localPart.split("/").filter(function(d) {return d.length > 0});
-                    localPart = localPart[localPart.length-1];
-                    getLabel.labels[uri] = localPart;
-                    promises[uri] = $q.defer();
-                    $http.get(ROOT_URL+'about?uri='+encodeURI(uri)+"&view=label")
-                        .then(function(data, status, headers, config) {
-                            if (status == 200) {
-                                var label = data.data;
-                                getLabel.labels[uri] = data.data;
-                                promises[uri].resolve(getLabel.labels[uri]);
-                            }
-                        });
-                }
-            }
-            return promises[uri].promise;
-        };
-        getLabel.labels = {};
-        return getLabel;
-    }]);
 
-    app.filter("label", ["$http", '$q','getLabel', function($http, $q, getLabel) {
-        function label(uri) {
-            if (getLabel.labels[uri] === undefined) {
-                var localPart = uri.split("#").filter(function(d) {return d.length > 0});
+    // Returns a promise that resolves to the label
+    app.factory("getLabel", ["$http", function($http) {
+	function getLabel(uri) {
+	    if(getLabel.labels[uri] === undefined) {
+		var localPart = uri.split("#").filter(function(d) {return d.length > 0});
                 localPart = localPart[localPart.length-1];
                 localPart = localPart.split("/").filter(function(d) {return d.length > 0});
                 localPart = localPart[localPart.length-1];
-                label.labels[uri] = {"label": localPart};
-                $http.get(ROOT_URL+'about?uri='+encodeURI(uri)+"&view=label")
-                    .then(function(data, status, headers, config) {
-                        if (status == 200) {
-                            var label = data.data;
-                            getLabel.labels[uri].label = data.data;
-                        }
-                    });
-            }
-            return getLabel.labels[uri];
-        };
-        label.labels = {};
-        return label;
+                getLabel.labels[uri] = { label: localPart };
+
+		getLabel.labels[uri].promise = $http.get(ROOT_URL+"about", {params: {uri, view: "label"}, responseType: "text"})
+						    .then(function(response) {
+							if(response.status === 200)
+							    getLabel.labels[uri].label = response.data;
+							return getLabel.labels[uri].label;
+						    });
+	    }
+	    return getLabel.labels[uri].promise;
+	}
+	getLabel.labels = {};
+	return getLabel;
     }]);
-    
+
+    app.filter("label", ["getLabel", function(getLabel) {
+	function label(uri) {
+	    if(getLabel.labels[uri] === undefined) {
+		getLabel(uri);
```

Comment:
It's a bit hard to see, with how Github is showing changes, but the logic here before was repeated in the getLabel factory, so I called getLabel() here instead, seeing as the filter was dependent on it anyway.